app/components/primer/beta/flash.rb

```diff
@@ -3,68 +3,61 @@
+      #     <%= render(Primer::Beta::Flash.new(scheme: :success)) { "This is a success flash message!" } %>
+      #   </div>
       #
       # @example Full width
       #   <%= render(Primer::Beta::Flash.new(full: true)) { "This is a full width flash message!" } %>
       #
       # @example Dismissible
-      #   <%= render(Primer::Beta::Flash.new(dismissible: true)) { "This is a dismissible flash message!" } %>
+      #   <%= render(Primer::Beta::Flash.new(dismissible: true, reappear: true)) { "This is a dismissible flash message!" } %>
       #
-      # @example Icon
+      # @example Custom icon
       #   <%= render(Primer::Beta::Flash.new(icon: :people)) { "This is a flash message with an icon!" } %>
       #
-      # @example With actions
+      # @example With action button
       #   <%= render(Primer::Beta::Flash.new) do |component| %>
       #     This is a flash message with actions!
-      #     <% component.with_action do %>
-      #       <%= render(Primer::ButtonComponent.new(size: :small)) { "Take action" } %>
+      #     <% component.with_action_button(size: :small) { "Take action" } %>
+      #   <% end %>
+      #
+      # @example With custom action
+      #   <%= render(Primer::Beta::Flash.new) do |component| %>
+      #     Comment saved!
+      #     <% component.with_action_content do %>
+      #       <%= render(Primer::IconButton.new(icon: :pencil, mr: 1, "aria-label": "Edit")) %>
       #     <% end %>
       #   <% end %>
       #
-      # @param full [Boolean] Whether the component should take up the full width of the screen.
-      # @param spacious [Boolean] Whether to add margin to the bottom of the component.
-      # @param dismissible [Boolean] Whether the component can be dismissed with an X button.
-      # @param icon [Symbol] Name of Octicon icon to use.
-      # @param scheme [Symbol] <%= one_of(Primer::Beta::Flash::SCHEME_MAPPINGS.keys) %>
-      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-      def initialize(full: false, spacious: false, dismissible: false, icon: nil, scheme: DEFAULT_SCHEME, **system_arguments)
-        @icon = icon
-        @dismissible = dismissible
-        @system_arguments = deny_tag_argument(**system_arguments)
-        @system_arguments[:tag] = :div
-        @system_arguments[:classes] = class_names(
-          @system_arguments[:classes],
-          "flash",
-          SCHEME_MAPPINGS[fetch_or_fallback(SCHEME_MAPPINGS.keys, scheme, DEFAULT_SCHEME)],
-          "flash-full": full
-        )
-        @system_arguments[:mb] ||= spacious ? 4 : nil
+      # @param system_arguments [Hash] The same arguments as <%= link_to_component(Primer::Beta::Banner) %>.
+      def initialize(**system_arguments)
+        super
+
+        @message_arguments[:aria] ||= {}
+        @message_arguments[:aria][:atomic] = "true"
+        @message_arguments[:role] = @scheme == :danger ? "alert" : "status"
```

Comment:
Here, I notice that we're setting Flash of type `:danger` to `role="alert"`,  and the other types of flash to `role="status"`. 

Since Flashes are displayed as a result of some user action, I think it could be argued that all flashes (whether it be of type `:danger`, `:success`, or `:warning`) are important enough to have `role="alert"` .

But, I can also understand the logic behind this update. 

Tagging @jscholes for thoughts on if we should distinguish what role to set based on the type of Flash.

Reply:
@khiga8 I think it is semantically correct to distinguish between roles.  However:

1. `role="status"` would need to be tested, in a similar way to how you tested your approach to alerts on dotcom.  It doesn't, AFAIK, have a corresponding system role like an alert, within the accessibility APIs on native operating systems.  So it's more of a standard live region.
2. Given that GitHub doesn't use alert that much, I don't think using it for all flashes would be inappropriate, if the testing fails/you don't have the cycles for it.

Reply:
I was able to test this and can confirm `role="status"` works as well as `role="alert"`.

Reply:
@camertron What screen readers and browsers did you test with, specifically?

Reply:
On Mac I tested with VoiceOver on Chrome, Safari, and Firefox. On Windows I tested with NVDA on Edge, Chrome, and Firefox.